plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/reader/TrinoReaderContext.java

```diff
@@ -0,0 +1,159 @@
+package io.trino.plugin.hudi.reader;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.hudi.avro.AvroSchemaUtils;
+import org.apache.hudi.common.engine.RecordContext;
+import org.apache.hudi.common.model.HoodieAvroIndexedRecord;
+import org.apache.hudi.common.model.HoodieEmptyRecord;
+import org.apache.hudi.common.model.HoodieKey;
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.table.HoodieTableConfig;
+import org.apache.hudi.common.table.read.BufferedRecord;
+import org.apache.hudi.common.util.AvroJavaTypeConverter;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+
+public class TrinoReaderContext
```

Comment:
Is this class mostly similar to `AvroRecordContext`?

Reply:
Could we directly use `AvroRecordContext`?  Once the merging logic is based on `Page` we can reimplement a new reader/record context.